tests/test_main.py

```diff
@@ -0,0 +1,44 @@
+import unittest
+from main import *
+from tests.json_test_data import *
+
+class TestMain(unittest.TestCase):
+    def test_parse_path_parts(self):
+        firestore_payload_name = "projects/river-sky-386919/databases/(default)/documents/untrusted_samples/fd1239d5488215a23440"
+        actual_collection_path, actual_document_path = parse_path_parts(firestore_payload_name)
+        self.assertEqual(actual_collection_path, "untrusted_samples")
+        self.assertEqual(actual_document_path, "fd1239d5488215a23440")
+
+    def test_is_document_creation_or_update_with_new_inputs___document_creation(self):
+        old_value = None
+        new_value = {
+            "test": "test"
+        }
+        result = is_document_creation_or_update_with_new_inputs(old_value, new_value)
+        self.assertTrue(result)
+
+    def test_is_document_creation_or_update_with_new_inputs___document_update_with_only_outputs_changed(self):
+        old_value = {
+            "test": "test"
+        }
+        new_value = {
+            "test": "test",
+            "validity_details": "test"
+        }
+        result = is_document_creation_or_update_with_new_inputs(old_value, new_value)
+        self.assertFalse(result)
+
+    def test_is_document_creation_or_update_with_new_inputs___document_update_with_new_inputs(self):
```

Comment:
split into a test per decorator.
Also, then we can have each decorator add its own html/pdf snippet.

So a report decorator will have:
* a value decorator
* unit tests
* pdf/html snippet generator

Reply:
These are just filters to run the validation though, so i think it makes sense to keep them in main and therefore in the main test. WDYT?